_posts/2024-08-02-gsoc2024_talagayev.md

```diff
@@ -0,0 +1,193 @@
+- Apply `Chem.RemoveHs` and `AllChem.Compute2DCoords` to obtain a 2D visualization of the molecule
+- Display the molecule via `IpyWidgets` with different checkboxes, that by selection would highlight specific parts of the molecule
+
+After discussion with the mentors and suggestions regarding how the code should be structured, it was decided that I would proceed with this suggestion and would apply the following steps with later on adding the additional
+features in the `IpyWidgets` visualization, which would give the user important information in regards to the characteristics of the molecule that is being visualized.
+
+## What I did
+
+I would divide this section into separate parts, each highlighting one of the important aspects of the work.
+
+### Main Visualization Code
+
+The main part of the code consisted of setting up the code for the visualization, which would allow the display of the molecule in a jupyter notebook. This is performed in the 
+`mdonatello.py`, which is mainly responsible for the visualization with it having the `class MoleculeVisualizer` that is the main class and responsible for the visualization with
+the functions in this class initializing the `IpyWidgets` interactive widgets such as a bar, that allows the user to select the molecule that he wants to display and also the
+interactive checkboxes with each having a feature, that I would explain in a separate section.
+
+This main code was the first part of the code that I designed and was in the beginning only able to display the molecule and had only one interactive checkbox, which displayed the atom
+indices of the molecules, but as more and more features were added the visualization grew bigger and had more features that it needed to display.
+
+### Feature Addition
+
+The feature addition was the next step that I was working on and was mainly the biggest part of the project since there was always an additional interesting feature that could be added
+that was proposed either by myself or one of the mentors.
+
+The first feature addition that I was working on were the physiochemical properties as well as the number of hydrogen bond donors and acceptors of the molecule.
+
+After it was done I started adding features, which would by selecting the checkbox highlight the features mentioned in the checkbox. For example, I implemented pharmacophore checkboxes,
+which by selection would highlight the selected pharmacophores, thus by selecting a hydrophobic feature, the atoms of the molecule that are hydrophobic via the `RDKit` pharmacophore
+recognition were highlighted in the assigned colors.
+
+Here is a short summary of features that I added for the visualization, which would show a value for the corresponding feature or adjust the figure of the molecule to display the values:
+
+- **Phyisochemical Properties**, **Atom Indices**, **Partial Charges**, **Hydrogen Bond Donors&Acceptors**
```

Comment:
I'd split these items on a new bullet point for each, same below

Reply:
I was worried that the blog would be too big at some point, since I have now 200 lines and thought that it could go easily above 250/300, so this is why I put them each close to each other, since I wasn't sure if there is some limit for the blog size :)